lib/nokogiri/xml/node.rb

```diff
@@ -192,16 +198,47 @@ def decorate!
 
       # :section: Manipulating Document Structure
 
-      ###
-      # Add +node_or_tags+ as a child of this Node.
-      #
-      # +node_or_tags+ can be a Nokogiri::XML::Node, a ::DocumentFragment, a ::NodeSet, or a String
-      # containing markup.
-      #
-      # Returns the reparented node (if +node_or_tags+ is a Node), or NodeSet (if +node_or_tags+ is
-      # a DocumentFragment, NodeSet, or String).
-      #
-      # Also see related method +<<+.
+      # :call-seq:
+      #   add_child(object) -> object
+      #
+      # Appends one or more children (as specified by the given +object+) to +self+;
+      # returns +object+.
+      #
+      # When +object+ is an XML::Node of any kind,
+      # adds the node as the last child of +self+;
+      # sets the parent of that node as +self+:
+      #
+      #   doc = XML::Document.new
+      #   doc.children.size      # => 0
+      #   root_ele = XML::Element.new('root', doc)
+      #   root_ele.parent        # => nil
+      #   doc.add_child(root_ele)
```

Comment:
I think this example should show adding a child to the root node, and not adding a root node to the parent.

Reply:
Fixed.